dist/index.js

```diff
@@ -85,25 +86,27 @@ exports['default'] = function () {
     protractor.stdout.on('data', function (buffer) {
       var text = buffer.toString();
       logger.protractor(text);
-      output = output + text;
     });
 
     protractor.stderr.on('data', function (buffer) {
       var text = buffer.toString();
       logger.protractor(text);
-      output = output + text;
     });
 
-    protractor.on('exit', function (status) {
-      handleTestEnd(status, output);
+    protractor.on('exit', function (code, signal) {
+      logger.info('Exited attempt ' + testAttempt + ' with code: ' + code + ' and signal: ' + signal + '\n');
+      handleTestEnd(code, '');
     });
 
     protractor.on('error', function (err) {
-      logger.log('info', 'Protractor failed to spawn ' + err + '\n', true);
+      logger.info('Protractor error ' + err + '\n', true);
+      handleTestEnd(1, 'Protractor error ' + err + '\n');
     });
   }
 
   if (testAttempt > 1 && testAttempt <= parsedOptions.maxAttempts) {
+    // We increment testAttempt in rerunFailedTests, so move it back first
+    --testAttempt;
```

Comment:
Why are we altering it here ? This seems confusing. Shouldn't rerun handle this ?

Reply:
This is to get the testAttempt right when used from commandline.